pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -673,6 +672,10 @@ func (m *kubeGenericRuntimeManager) executePreStopHook(ctx context.Context, pod
 	case <-done:
 		klog.V(3).InfoS("PreStop hook completed", "pod", klog.KObj(pod), "podUID", pod.UID,
 			"containerName", containerSpec.Name, "containerID", containerID.String())
+	case <-ctx.Done():
```

Comment:
It's no need to add `ctx.Done()` here. If ctx is cancelled, `m.runner.Run` will finish and `done` will be closed.

Reply:
No, if you don't go to select, it won't execute the exit.

Reply:
> No, if you don't go to select, it won't execute the exit.

Why? `m.runner.Run` will stop if ctx is cancelled and `case <- done` will be selected.